### PR TITLE
Added C# code examples for Using signals page.

### PR DESCRIPTION
**Pull request by sbalia**
Also:
* changed a bit one GDscript example.
* added one macOS shortcut.

### CODE REVIEW

File: getting_started/step_by_step/signals.rst [hunk 1]
```diff
@@ -255,6 +300,14 @@ We can now connect the Timer to the Sprite2D in the ``_ready()`` function.
         var timer = get_node("Timer")
         timer.timeout.connect(_on_Timer_timeout)
 
+ .. code-tab:: csharp C#
+
+    public override void _Ready()
+    {
+        var timer = GetNode<Timer>("Timer");
+        timer.Connect("timeout", this, "OnTimerTimeout");
```
Comment (raulsntos) [inline getting_started/step_by_step/signals.rst:308]: We recommend using `nameof` for the method name as it's less error-prone.

```suggestion
        timer.Connect("timeout", this, nameof(OnTimerTimeout));
```

File: getting_started/step_by_step/signals.rst [hunk 2]
```diff
@@ -326,6 +419,18 @@ reaches 0.
...
+ .. code-tab:: csharp C#
+
+    using Godot;
+
+    public class CustomSignal : Node2D
+    {
+        [Signal]
+        public delegate void HealthDepleted();
+
+        private int Health = 10;
```
Comment (raulsntos) [inline getting_started/step_by_step/signals.rst:431]: The C# naming convention for fields is camelCase.

```suggestion
        private int health = 10;
```

In the engine we also prefix private fields with an underscore `_` but I'm not sure if that also applies to the documentation.

File: getting_started/step_by_step/signals.rst [hunk 3]
```diff
@@ -344,6 +449,18 @@ To emit a signal in your scripts, call ``emit_signal()``.
...
 
+ .. code-tab:: csharp C#
+
+    public void TakeDamage(int amount)
+    {
+        Health -= amount;
+
+        if (Health < 0)
+        {
+            EmitSignal("HealthDepleted");
```
Comment (raulsntos) [inline getting_started/step_by_step/signals.rst:460]: We recommend using `nameof` for the method name as it's less error-prone.

```suggestion
            EmitSignal(nameof(HealthDepleted));
```

File: getting_started/step_by_step/signals.rst [hunk 4]
```diff
@@ -354,6 +471,20 @@ names between parentheses:
...
+ .. code-tab:: csharp C#
+
+    using Godot;
+
+    public class CustomSignal : Node
+    {
+        [Signal]
+        public delegate void HealthChanged(int oldValue, int newValue);
+
+        private int Health = 10;
```
Comment (raulsntos) [inline getting_started/step_by_step/signals.rst:485]: The C# naming convention for fields is camelCase.

```suggestion
        private int health = 10;
```

In the engine we also prefix private fields with an underscore `_` but I'm not sure if that also applies to the documentation.

File: getting_started/step_by_step/signals.rst [hunk 5]
```diff
@@ -372,6 +503,15 @@ To emit values along with the signal, add them as extra arguments to the
         health -= amount
         emit_signal("health_changed", old_health, health)
 
+ .. code-tab:: csharp C#
+
+    public void TakeDamage(int amount)
+    {
+        var oldHealth = health;
+        Health -= amount;
+        EmitSignal("HealthChanged", oldHealth, Health);
```
Comment (raulsntos) [inline getting_started/step_by_step/signals.rst:512]: We recommend using `nameof` for the method name as it's less error-prone.

```suggestion
        EmitSignal(nameof(HealthDepleted), oldHealth, Health);
```

File: getting_started/step_by_step/signals.rst [hunk 6]
```diff
@@ -153,6 +153,13 @@ the ``not`` keyword to invert the value.
     func _on_Button_pressed():
         set_process(not is_processing())
 
+ .. code-tab:: csharp C#
+
+    public void OnButtonPressed()
+    {
+        SetProcess(!IsProcessing()); // "!" - means not.
```
Comment (raulsntos) [inline getting_started/step_by_step/signals.rst:160]: I don't think the comment is necessary.

File: getting_started/step_by_step/signals.rst (hunk 1 above)
Comment (sbalia) [reply getting_started/step_by_step/signals.rst]: done

File: getting_started/step_by_step/signals.rst (hunk 2 above)
Comment (sbalia) [reply getting_started/step_by_step/signals.rst]: Made it with a lower case letter, without an underscore sign. 

In my opinion, it's better to have it with `_`, but it won't be consistent with other documentation.

File: getting_started/step_by_step/signals.rst (hunk 3 above)
Comment (sbalia) [reply getting_started/step_by_step/signals.rst]: done

File: getting_started/step_by_step/signals.rst (hunk 5 above)
Comment (sbalia) [reply getting_started/step_by_step/signals.rst]: done

File: getting_started/step_by_step/signals.rst (hunk 6 above)
Comment (sbalia) [reply getting_started/step_by_step/signals.rst]: removed

File: getting_started/step_by_step/signals.rst (hunk 2 above)
Comment (sbalia) [reply getting_started/step_by_step/signals.rst:431]: Reverted back to the uppercase variant to match it with the current docs. As well as other variables, like Speed and AngularSpeed.

File: getting_started/step_by_step/signals.rst (hunk 4 above)
Comment (sbalia) [reply getting_started/step_by_step/signals.rst:485]: Reverted back to the uppercase variant to match it with the current docs. As well as other variables, like Speed and AngularSpeed.

**Review by raulsntos — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you! 🎉

**Pull request merged**